src/whitenoise/accounts.rs

```diff
@@ -857,10 +849,77 @@ impl Whitenoise {
+                Err(e) => {
+                    tracing::error!(
+                        target: "whitenoise::background_fetch_account_data",
+                        "Failed to get Whitenoise instance for background sync: {}",
+                        e
+                    );
+                    return;
+                }
+            };
+
+            // Get user's relay information for contact list fetching
+            let user_relays = match account_clone.nip65_relays(whitenoise).await {
+                Ok(relays) => relays.into_iter().map(|r| r.url).collect::<Vec<_>>(),
+                Err(_) => {
+                    // Fallback to default relays if user has no specific relays
+                    Relay::defaults()
+                        .into_iter()
+                        .map(|r| r.url)
+                        .collect::<Vec<_>>()
+                }
+            };
+
+            // Ensure user relays are connected before fetching contact list
+            if let Err(e) = whitenoise.nostr.ensure_relays_connected(&user_relays).await {
+                tracing::warn!(
+                    target: "whitenoise::background_fetch_account_data",
+                    "Failed to ensure relay connections for account {}: {}. Proceeding with fetch anyway.",
+                    account_clone.pubkey.to_hex(),
+                    e
+                );
+            }
+
+            // Fetch the account's contact list
+            let fetched_contact_list = whitenoise
+                .nostr
+                .fetch_contact_list_events(account_clone.pubkey, &user_relays)
+                .await;
+
+            let contact_list_pubkeys = match &fetched_contact_list {
+                Ok(Some(contact_list_event)) => pubkeys_from_event(contact_list_event),
+                Ok(None) => Vec::new(),
+                Err(e) => {
+                    tracing::error!(target: "whitenoise::background_fetch_account_data", "Failed to fetch contact list for account {}: {}", account_clone.pubkey.to_hex(), e);
+                    Vec::new()
+                }
+            };
+
+            // Process the fetched contact list first (if it exists)
+            // This ensures existing contact lists are processed during login even if no new events are streamed
+            if let Ok(Some(contact_list_event)) = fetched_contact_list {
+                tracing::debug!(
+                    target: "whitenoise::nostr_manager::sync_all_user_data",
+                    "Processing fetched contact list for account {}",
+                    account_clone.pubkey.to_hex()
+                );
+                if let Err(e) = whitenoise
+                    .handle_contact_list(&account_clone, contact_list_event)
+                    .await
+                {
+                    tracing::error!(target: "whitenoise::background_fetch_account_data", "Failed to process contact list for account {}: {}", account_clone.pubkey.to_hex(), e);
+                }
+            }
+
```

Comment:
I think we can move this to the `process_user_event_streams`, I'm not sure why it needs to be processed before fetching the other events